sensor/kubernetes/upgrade/command_handler_impl.go

```diff
@@ -136,8 +137,14 @@ func (h *commandHandler) ProcessMessage(msg *central.MsgToSensor) error {
 		return nil
 	}
 
-	if h.configHandler.GetHelmManagedConfig() != nil && !h.configHandler.GetHelmManagedConfig().GetNotHelmManaged() {
-		upgradesNotSupportedErr := errors.New("Cluster is Helm-managed and does not support auto-upgrades")
+	if h.configHandler.GetHelmManagedConfig() != nil {
+		var upgradesNotSupportedErr error
+		switch h.configHandler.GetHelmManagedConfig().GetManagedBy() {
+		case storage.ManagerType_MANAGER_TYPE_HELM_CHART:
+			upgradesNotSupportedErr = errors.New("Cluster is Helm-managed and does not support auto-upgrades")
+		case storage.ManagerType_MANAGER_TYPE_KUBERNETES_OPERATOR:
+			upgradesNotSupportedErr = errors.New("Cluster is Operator-managed and does not support auto-upgrades")
+		}
```

Comment:
Oh, hang on, I think this is might be wrong. In the old version, the `if` already checked the condition. Here we don't know until the switch block. But it looks like we delete upgrader etc either way!

I wonder whether this will get caught by any test...

Reply:
That is a pretty good catch! I addressed it and will check if there are any tests that could cover that scenario.

Reply:
There are no tests that would cover that directly. There is a chance that the existing test `Test sensor cert rotation with upgrader succeeds for manifest based installed clusters` would catch that. I would need to write something completely new for this scenario, but that is unrealistic given the time left to code freeze. Do you agree to merge it as it is and write the test in a follow-up PR?